apps/www/content/docs/components/spinner.mdx

```diff
@@ -0,0 +1,66 @@
+---
+title: Spinner
+description: Renders a loading indicator.
+component: true
+---
+
+<ComponentPreview name="spinner-demo" />
+
+## About
+
+The Spinner is built and maintained by [flixlix](https://luca-felix.com).
+
+## Installation
+
+<Tabs defaultValue="cli">
+
+<TabsList>
+  <TabsTrigger value="cli">CLI</TabsTrigger>
+  <TabsTrigger value="manual">Manual</TabsTrigger>
+</TabsList>
+<TabsContent value="cli">
+
+<Steps>
+
+<Step>Run the following command:</Step>
+
+```bash
+npx shadcn-ui@latest add spinner
+```
+
+<Step>Add the `spinner` animation to your tailwind config</Step>
+```js title="tailwind.config.js" {9-12, 16}
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  ...
+  theme: {
+    extend: {
+      ...
+      keyframes: {
+        ...
+        "spinner": {
+          "0%": { opacity: 1 },
+          "to": { opacity: .15 }
```

Comment:
> @flixlix looks like something wrong here
> 
> ![image](https://private-user-images.githubusercontent.com/142723369/329756095-7ae938e5-9100-47ba-90c9-caa572073fa4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU0MDU3NTUsIm5iZiI6MTcxNTQwNTQ1NSwicGF0aCI6Ii8xNDI3MjMzNjkvMzI5NzU2MDk1LTdhZTkzOGU1LTkxMDAtNDdiYS05MGM5LWNhYTU3MjA3M2ZhNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxMVQwNTMwNTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTNhNTcxMmQ5MTI0OTM3YjcxM2YwZDI0MjJiZWMwYWQ0YmMwOGY0NWJiMjY1YzJhOWQ3YzM5MWRiMTAxYTQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Lym45BLbxK_q-mdodnL3-sNfDBlVSedj56vC4Nnswg0)
> 
> and can you double check this
> 
> ```json
> ==> "0%": 
> ==> "to":
> ```
> 
> I think I should be
> 
> ```json
> spinner: {
>   from: { opacity: 1 },
>   to: { opacity: 0.15 },
> },
> ```
> 
> but still I see the TS error `Type 'number' is not assignable to type 'string'.ts(2322)`
> 
> this is just a TS error the component works perfectly fine.
> 
> ![image](https://private-user-images.githubusercontent.com/142723369/329756317-c829cf8b-8388-4b26-87d8-27b26abe5dd6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU0MDU3NTUsIm5iZiI6MTcxNTQwNTQ1NSwicGF0aCI6Ii8xNDI3MjMzNjkvMzI5NzU2MzE3LWM4MjljZjhiLTgzODgtNGIyNi04N2Q4LTI3YjI2YWJlNWRkNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxMVQwNTMwNTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTI4MjFjMTBhNDNmZDZkNzAxNDZkMmZkMGQzODRjMzdiZmViZjhlYmFjZTE5MGIxZTkyNGQ2ZDZiNjQ3MzgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.zZIr0Fb9a3czIMdpaDRhiZdEdoIOijKA4dxck8LOcN0)

Reply:
@flixlix @ruru-m07 

> but still I see the TS error Type 'number' is not assignable to type 'string'.ts(2322)

```ts
spinner: {
    from: { opacity: "1" },
    to: { opacity: "0.15" },
}
```

Enclosing numbers in double quotation marks will eliminate the TypeScript error.

Reply:
fixed both issues